cli/config/configfile/file_test.go

```diff
@@ -157,3 +194,90 @@ func TestGetAllCredentials(t *testing.T) {
 	expected["example.com/foo"] = exampleAuth
 	assert.Equal(t, expected, authConfigs)
 }
+
+func TestGetAllCredentialsCredsStore(t *testing.T) {
+	configFile := New("filename")
+	configFile.CredentialsStore = "test_creds_store"
+	testServerAddress := "example.com"
+	expectedAuth := types.AuthConfig{
+		Username: "user",
+		Password: "pass",
+	}
+
+	newNativeStore = func(configFile *ConfigFile, helperSuffix string) credentials.Store {
+		return NewMockNativeStore(map[string]types.AuthConfig{testServerAddress: expectedAuth})
+	}
+
+	authConfigs, err := configFile.GetAllCredentials()
+	require.NoError(t, err)
+
+	expected := make(map[string]types.AuthConfig)
+	expected[testServerAddress] = expectedAuth
+	assert.Equal(t, expected, authConfigs)
+}
+
+func TestGetAllCredentialsCredHelper(t *testing.T) {
+	testFileStoreServerAddress := "example.com"
+	testCredHelperSuffix := "test_cred_helper"
+	testCredHelperServerAddress := "credhelper.com"
+
+	expectedFileStoreAuth := types.AuthConfig{
+		Username: "file_store_user",
+		Password: "file_store_pass",
+	}
+	expectedCredHelperAuth := types.AuthConfig{
+		Username: "cred_helper_user",
+		Password: "cred_helper_pass",
+	}
+
+	configFile := New("filename")
+	configFile.CredentialHelpers = map[string]string{testCredHelperServerAddress: testCredHelperSuffix}
+	configFile.AuthConfigs["https://"+testFileStoreServerAddress] = expectedFileStoreAuth
+
+	newNativeStore = func(configFile *ConfigFile, helperSuffix string) credentials.Store {
+		return NewMockNativeStore(map[string]types.AuthConfig{testCredHelperServerAddress: expectedCredHelperAuth})
+	}
+
+	authConfigs, err := configFile.GetAllCredentials()
+	require.NoError(t, err)
+
+	expected := make(map[string]types.AuthConfig)
+	expected[testFileStoreServerAddress] = expectedFileStoreAuth
+	expected[testCredHelperServerAddress] = expectedCredHelperAuth
+	assert.Equal(t, expected, authConfigs)
+}
+
+func TestGetAllCredentialsCredStoreAndCredHelper(t *testing.T) {
```

Comment:
I think this one test might cover all the behaviour, but I guess it doesn't hurt to have the other ones. They should be pretty fast.